File: test/aidbox_sdk/snapshots/hl7-fhir-r4-core/Subscription.cs
using Aidbox.FHIR.Base;
using Aidbox.FHIR.Utils;

namespace Aidbox.FHIR.R4.Core;

public class Subscription : DomainResource, IResource
{
    public string? End { get; set; }
    public string? Error { get; set; }
    public required string Reason { get; set; }
    public required string Status { get; set; }
    public required Subscription_Channel Channel { get; set; }
    public Base.ContactPoint[]? Contact { get; set; }
    public required string Criteria { get; set; }

    public class Subscription_Channel : BackboneElement
    {
        public required string Type { get; set; }
        public string[]? Header { get; set; }
        public string? Payload { get; set; }
        public string? Endpoint { get; set; }
    }
}

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Subscription.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import ContactPoint
from base import DomainResource
from base import BackboneElement

@dataclass(kw_only=True)
class SubscriptionChannel(BackboneElement):
    type: str
    header: Optional[List[str]] = field(default_factory=list)
    payload: Optional[str] = None
    endpoint: Optional[str] = None

@dataclass(kw_only=True)
class Subscription(DomainResource):
    end: Optional[str] = None
    error: Optional[str] = None
    reason: str
    status: str
    channel: SubscriptionChannel
    contact: Optional[List[ContactPoint]] = field(default_factory=list)
    criteria: str